src/Features/Core/Portable/CodeRefactorings/ConvertToInterpolatedString/AbstractConvertToInterpolatedStringRefactoringProvider.cs

```diff
@@ -0,0 +1,159 @@
+﻿using System;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.CodeRefactorings
+{
+    internal abstract class AbstractConvertToInterpolatedStringRefactoringProvider<TInterpolatedStringExpressionSyntax, TInvocationExpressionSyntax, TExpressionSyntax, TArgumentSyntax, TLiteralExpressionSyntax> : CodeRefactoringProvider
+        where TInterpolatedStringExpressionSyntax : SyntaxNode
+        where TInvocationExpressionSyntax : SyntaxNode
+        where TExpressionSyntax : SyntaxNode
+        where TArgumentSyntax : SyntaxNode
+        where TLiteralExpressionSyntax : SyntaxNode
+    {
+        protected abstract SeparatedSyntaxList<TArgumentSyntax>? GetArguments(TInvocationExpressionSyntax invocation);
+        protected abstract ImmutableArray<TExpressionSyntax> GetExpandedArguments(SemanticModel semanticModel, SeparatedSyntaxList<TArgumentSyntax> arguments);
+        protected abstract TLiteralExpressionSyntax GetFirstArgument(SeparatedSyntaxList<TArgumentSyntax> arguments);
+        protected abstract TInterpolatedStringExpressionSyntax GetInterpolatedString(string text);
+        protected abstract string GetText(SeparatedSyntaxList<TArgumentSyntax> arguments);
+        protected abstract bool IsArgumentListCorrect(TInvocationExpressionSyntax invocation, ISymbol invocationSymbol, ImmutableArray<ISymbol> formatMethods, SemanticModel semanticModel, CancellationToken cancellationToken);
+        protected abstract bool IsStringLiteral(TLiteralExpressionSyntax firstArgument);
+        protected abstract TInterpolatedStringExpressionSyntax VisitArguments(ImmutableArray<TExpressionSyntax> expandedArguments, TInterpolatedStringExpressionSyntax interpolatedString);
+
+        public override async Task ComputeRefactoringsAsync(CodeRefactoringContext context)
+        {
+            var semanticModel = await context.Document.GetSemanticModelAsync(context.CancellationToken).ConfigureAwait(false);
+
+            var stringType = semanticModel.Compilation.GetTypeByMetadataName("System.String");
```

Comment:
I would use the SpecialType.System_String enum instead and call Compilation.GetSpecialType.